modules/GenericDataParser/src/server/generic_data_parser.R

```diff
@@ -3953,6 +4063,133 @@ getFormatParameters <- function(rawOnlyFormat, customFormatSettings, inputFormat
+          } else if (value[['lsKind']] == "column type") {
+            dataTypeEntry = value[['stringValue']]
+          }
+          
+        }
+      }
+
+      #add the endpoint to the dataframe
+      dataEntry <- list(columnName = columnNameEntry, units = unitsEntry, dataType= dataTypeEntry)
+      protocolEndpointDataFrame = rbind(protocolEndpointDataFrame, dataEntry, stringsAsFactors=FALSE)
+
+    }
+  }
+
+  return(protocolEndpointDataFrame)
+}
+
+validateExperimentColumns <- function(selColumnOrderInfo, protocolEndpointDataFrame) {
+  # Checks if the experiment columns names/units/data type are found in the protocol endpoint data
+  # Raises errors if an experiment column is not valid 
+  # Only used for protocols with strict endpoint enabled
+
+  for (experimentRowNum in seq(1, nrow(selColumnOrderInfo))) {
+    experimentRowData <- selColumnOrderInfo[experimentRowNum,]
+    experimentRowName = experimentRowData$valueKind
+    experimentRowUnits = experimentRowData$Units
+    experimentRowDataType = experimentRowData$valueType
+    
+    #the experiment column must match at least one endpoint in the protocol to pass 
+    experimentRowMatchesEndpoint = FALSE
+
+    # we need to check if the experiment row data matches any one of the protocol endpoints
+    for (protocolRowNum in seq(1,nrow(protocolEndpointDataFrame))) {
+      # if a protocol value is NA, we consider it a match (since any value is acceptable)
+      # if the experiment values also match the protocol values, we consider it a match
+      # all three (rowNamesMatch, rowUnitsMatch, and rowTypesMatch) must be true
+      protocolRowData <- protocolEndpointDataFrame[protocolRowNum,]
+
+      protocolRowName = protocolRowData$columnName
+      if (is.na(protocolRowName) | experimentRowName == protocolRowName) {
+        rowNamesMatch = TRUE 
+      } else {
+        rowNamesMatch = FALSE
+      }
+
+      protocolRowUnits = protocolRowData$units
+      if (is.na(protocolRowUnits) | experimentRowUnits == protocolRowUnits) {
+        rowUnitsMatch = TRUE 
+      } else {
+        rowUnitsMatch = FALSE
+      }
+
+      protocolRowDataType = protocolRowData$dataType
+      if (is.na(protocolRowDataType) | experimentRowDataType == protocolRowDataType) {
+        rowTypesMatch = TRUE 
+      } else {
+        rowTypesMatch = FALSE
+      }
+
+      #if the experiment column matches one of the endpoints for the protocol, record it and move onto the next column
+      if (rowNamesMatch & rowUnitsMatch & rowTypesMatch) {
+        experimentRowMatchesEndpoint = TRUE
+        break 
+      }
+
+    }
+
+    if (experimentRowMatchesEndpoint == FALSE) {
+      addError(paste0("The result type '", experimentRowName, "' with data type '", experimentRowDataType, "' and units '", experimentRowUnits, "' is not one of the allowed result types for this protocol. Please revise your file or contact an ACAS administrator to update the allowed result types for this protocol."))
```

Comment:
Here's where you'll want to split between error & warning based on the "strict"ness of the protocol.

See the last image on https://confluence.schrodinger.com/x/egSgBw for some proposed verbiage.